xunit-autolabeler-v2/ast_parser/lib/file_utils.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+from os import path
+import subprocess
+
+from .constants import REGION_TAG_GREP_ARGS
+
+
+def __getFiles(root_dir, predicate):
+    if path.basename(path.normpath(root_dir)).startswith('.'):
+        # Ignore dot-directories
+        return []
+
+    paths = [path.join(root_dir, p) for p in os.listdir(root_dir)
+             if p != 'lib']
+
+    folders = [p for p in paths if not os.path.isfile(p)]
+
+    files = [p for p in paths if os.path.isfile(p)
+             and predicate(os.path.basename(p))]
+
+    for f in folders:
+        files += __getFiles(f, predicate)
+
+    return files
+
+
+def get_python_files(root_dir):
+    return __getFiles(root_dir, lambda p: p.endswith('.py'))
+
+
+def get_yaml_files(root_dir):
```

Comment:
What does this function do?

Reply:
Per the design doc, `.drift-data.yml` files are used to store (manually maintained) metadata on snippets. (For example, they're useful for snippets that aren't automatically detected by the AST parser.)